test/functional/terminal/helpers.lua

```diff
@@ -4,19 +4,28 @@
 local helpers = require('test.functional.helpers')(nil)
 local Screen = require('test.functional.ui.screen')
 local testprg = helpers.testprg
+local exec_lua = helpers.exec_lua
 local feed_command, nvim = helpers.feed_command, helpers.nvim
 
 local function feed_data(data)
-  -- A string containing NUL bytes is not converted to a Blob when
-  -- calling nvim_set_var() API, so convert it using Lua instead.
-  nvim('exec_lua', 'vim.g.term_data = ...', {data})
-  nvim('command', 'call jobsend(b:terminal_job_id, term_data)')
+  exec_lua('vim.api.nvim_chan_send(vim.b.terminal_job_id, ...)', data)
```

Comment:
`chansend()` (`jobsend()`) can accept either a list or a string, while `nvim_chan_chan` only accepts a string, so `table.concat` is needed here when `data` is a table.